File: scripts/test_apis.py
import os
from dotenv import load_dotenv
from pinecone import Pinecone
import google.generativeai as genai

# Load env
load_dotenv("server/.env")

print("Key Check:")
print(f"GEMINI_API_KEY: {'Loaded' if os.getenv('GEMINI_API_KEY') else 'Not Found'}")
print(f"PINECONE_API_KEY: {'Loaded' if os.getenv('PINECONE_API_KEY') else 'Not Found'}")
print("-" * 30)

# Check Gemini
print("Testing Gemini API...")
try:
    genai.configure(api_key=os.getenv("GEMINI_API_KEY"))
    # test a simple model call
    model = genai.GenerativeModel("gemini-1.5-flash")
    response = model.generate_content("Say 'Hello'")
    if response and response.text:
        print(f"✅ Gemini API is Working! Response: '{response.text.strip()}'")
    else:
        print("❌ Gemini API returned empty response.")
except Exception as e:
    print(f"❌ Gemini API Error: {e}")

# Check Pinecone
print("-" * 30)
print("Testing Pinecone API...")
try:
    pc = Pinecone(api_key=os.getenv("PINECONE_API_KEY"))
    indexes = pc.list_indexes()
    index_names = [idx.name for idx in indexes]  # Adjusted depending on pinecone client version
    print(f"✅ Pinecone API is Working! Total Indexes listed: {len(index_names)}")
    if index_names:
        print(f"Indexes: {index_names}")
    
    index_name = os.getenv("PINECONE_INDEX_NAME")
    if index_name in index_names:
        index = pc.Index(index_name)
        stats = index.describe_index_stats()
        print(f"✅ Index '{index_name}' is accessible. Stats: {stats}")
    else:
        print(f"⚠️ Index '{index_name}' configured in .env but not found in your Pinecone account.")
except Exception as e:
    print(f"❌ Pinecone API Error: {e}")

File: scripts/test_apis.js
const fs = require('fs');
const path = require('path');

// Simple dotenv parser
const envPath = path.join(__dirname, '../server/.env');
console.log(`Reading env from ${envPath}`);
const env = fs.readFileSync(envPath, 'utf8')
  .split('\n')
  .reduce((acc, line) => {
    const match = line.match(/^\s*([\w.-]+)\s*=\s*(.*)?\s*$/);
    if (match) {
      let val = match[2] || '';
      val = val.replace(/^['"](.*)['"]$/, '$1').trim();
      acc[match[1]] = val;
    }
    return acc;
  }, {});

const geminiKey = env.GEMINI_API_KEY;
const pineconeKey = env.PINECONE_API_KEY;
const pineconeIndex = env.PINECONE_INDEX_NAME;

console.log('Keys loaded:', { 
  gemini: !!geminiKey, 
  pinecone: !!pineconeKey,
  index: pineconeIndex
});

async function testGemini() {
  console.log('\n--- Testing Gemini API ---');
  try {
    const url = `https://generativelanguage.googleapis.com/v1beta/models/gemini-pro:generateContent?key=${geminiKey}`;
    const response = await fetch(url, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        contents: [{ parts: [{ text: "Respond with exactly the word 'OK'." }] }]
      })
    });
    const data = await response.json();
    if (data.error) throw new Error(data.error.message);
    const text = data.candidates?.[0]?.content?.parts?.[0]?.text;
    console.log('✅ Gemini API is Working! Response:', (text || '').trim());
  } catch (err) {
    console.error('❌ Gemini API Error:', err.message);
  }
}

async function testPinecone() {
  console.log('\n--- Testing Pinecone ---');
  try {
    const url = 'https://api.pinecone.io/indexes';
    const response = await fetch(url, {
      headers: {
        'Api-Key': pineconeKey
      }
    });
    const data = await response.json();
    if (data.error) throw new Error(JSON.stringify(data.error));
    
    const indexList = data.indexes ? data.indexes.map(i => i.name) : [];
    console.log(`✅ Pinecone API Working! Discovered indexes: ${indexList.join(', ') || 'None'}`);
    
    if (indexList.includes(pineconeIndex)) {
      console.log(`✅ Required index '${pineconeIndex}' is present!`);
      const indexInfo = data.indexes.find(i => i.name === pineconeIndex);
      if (indexInfo?.host) {
        const statsUrl = `https://${indexInfo.host}/describe_index_stats`;
        const statsRes = await fetch(statsUrl, {
          method: 'POST',
          headers: { 'Api-Key': pineconeKey, 'Content-Type': 'application/json' }
        });
        if (statsRes.ok) {
           const stats = await statsRes.json();
           console.log(`✅ Index Stats for '${pineconeIndex}': Total vectors: ${stats.totalVectorCount}, Dimension: ${stats.dimension}`);
        } else {
           console.log(`⚠️ Could not fetch stats: ${statsRes.status}`);
        }
      }
    } else {
      console.log(`⚠️ Index '${pineconeIndex}' not found in your Pinecone account.`);
    }

  } catch (err) {
    console.error('❌ Pinecone API Error:', err.message);
  }
}

async function run() {
  await testGemini();
  await testPinecone();
}

run();
